sphinx/writers/latex.py

```diff
@@ -1203,6 +1204,12 @@ def depart_table(self, node):
             endmacro = '\\end{tabulary}\n\n'
         if self.table.colspec:
             self.body.append(self.table.colspec)
+        elif self.table.colwidths:
+            total = sum(self.table.colwidths)
+            colspec = ['p{\\dimexpr(\\linewidth-\\arrayrulewidth)*%d/%d'
+                       '-2\\tabcolsep-\\arrayrulewidth\\relax}' % (width, total)
```

Comment:
It is possible to do something like 

    \newcolumntype{X}[2]{p{\dimexpr(\linewidth-\arrayrulewidth)*#1/#2-2\tabcolsep-\arrayrulewidth\relax}}

in sphinx.sty and then generate mark-up by writer like 

    |X{50}{70}|X{20}{70}|

for smaller files. I must check which letter is best available for this (i.e. not used by packages user may add to default config).

Fortunately `\dimexpr` handles `<dimen>*A/B` in double precision for the first multiplication so that even with things like `500/1000` there will be no overflow, despite the maximal dimension in TeX being only about `5.76 m`.